### PR TITLE
[App Components] Fix flow for components

### PR DESCRIPTION
**Pull request by loui981c**
## Description

Fixing the order and functionality of the flow between components.

## Changes

Fetching the different section components in the same order as the content-creator has created them. It is impossible to go to the next slide if the slide is an exercise and the user has not answered the exercise. When the user stands on the last slide, the continue button navigates to completeSection. 

## Checklist

- [X] Code has been tested locally and passes all relevant tests.
- [X] Documentation has been updated to reflect the changes, if applicable.
- [X] Code follows the established coding style and guidelines of the project.
- [X] All new and existing tests related to the changes have passed.
- [X] Any necessary dependencies or new packages have been properly documented.
- [X] Pull request title and description are clear and descriptive.
- [X] Reviewers have been assigned to the pull request.
- [X] Any potential security implications have been considered and addressed.
- [X] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)

Isn't really necessary for this PBI 

## If mobile/frontend pull request, what version of the backend is it stable, and running on? 

Branch: course_gamified_points

Commit id: 61b1be3

## Notes for Reviewers

The last section 'Section with lectures and exercises' of the course 'Financais Pessoais' should work since it is the only section with the new structure in DB

### CODE REVIEW

File: eml/components/exercise/CoursePoints.jsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+          setTimeout(() => {
+            setScale(1);
+          }, 75 * counter);
+          return finalValue;
+        }
+        return nextNumber;
+      });
+    }, 50);
+
+    // TODO: brokey, magnus help
```
Comment (JackSMunn) [inline eml/components/exercise/CoursePoints.jsx:44]: :D

**Review by JackSMunn — CHANGES_REQUESTED** [review]

**Review by JackSMunn — APPROVED** [review]

**Comment by JackSMunn** [comment]
Gj, did not test locally though.

File: eml/components/exercise/CoursePoints.jsx (hunk 1 above)
Comment (PrometheusBork) [reply eml/components/exercise/CoursePoints.jsx:44]: + 1

**Review by ITPini — APPROVED** [review]

File: eml/api/api.js [hunk 2]
```diff
@@ -279,6 +278,19 @@ export const getLectureById = async (lectureId) => {
 
 };
 
+export const getExerciseById = async (exerciseId) => {
+  try {
```
[the remark this answers is not in the document]
Comment (loui981c) [reply eml/api/api.js:282]: a mistake but i have already merged since 2 approved. will fix in next pbi

**Pull request merged**